### DISCUSSION TITLE
Paperless-ngx connections to NFS shares

### DISCUSSION THREAD

**Discussion by FrankDUS69** [tteck/Proxmox#619] [Help Needed]
I´ve used the script to create the paperless-ngx container, it worked flawlessly.

But I struggle to connext the consume and media/documents folder with my NFS share on my Synology (my Diskstation doesn´t support VM or Docker).

I have created a shared folder "Inbox" and "Archiv" on my Diskstation and gave permissions to paperless IP. Then I created mount points via <LXC>.conf in PVE. 
![paperless MP](https://user-images.githubusercontent.com/87012598/194839164-dfcd015f-d520-4355-a33d-532addad2e64.jpg)
But I don´t know how to connect them to the consume or media folder. I guess it should be done via paperless.conf, but I´m unsure how this works. 
Maybe it sounds like a stupid beginners question, but I am quite new in this LINUX theme. Maybe someone can help with an example configuration?

**Comment by tomdh76** [comment]
If you are in the LXC container can you see the contents of /mnt/pve/inbox? 
If so then you only have to change in the paperless.conf:

PAPERLESS_CONSUMPTION_DIR=/mnt/pve/inbox (assuming this is the folder where you put in new documents)

PAPERLESS_MEDIA_ROOT=/mnt/pve/archiv (this will be the folder where all documents are stored after consuming.

**Reply by FrankDUS69** [reply]
Unfortunately not. The folder doesn´t exist (?).
Meanwhile I pointed the mount points as follows:
![grafik](https://user-images.githubusercontent.com/87012598/199539682-e9009d71-6851-4e59-a4e4-ab10010a32f7.png)

When I create a test.txt file in /opt/paperless/consume, I see it on my NAS in "Inbox", but when I add a new file on the NAS, I can´t see it in the LXC. Any idea?

**Reply by tomdh76** [reply]
And in the PVE shell? Can you see the file in /mnt/pve/Inbox? Have you give permissions in the NAS to the ip of proxmox?

**Reply by FrankDUS69** [reply]
I see all files of the NFS share in the PVE shell. Permissions of NFS share are given to both PVE and LXC IP. All users on the NAS are squashed to admin actually. 
Afaik you can´t add NFS storage in Proxmox UI without permissions.

**Comment by Arkheon** [comment]
hoi @FrankDUS69 ,

On your LXC try this :

```
apt update && apt install nfs-common -y
```

Then modified /etc/fstab and added this 

```
192.168.1.234:/volume1/Documents/consume /opt/paperless/consume nfs vers=3,nouser,atime,auto,retrans=2,rw,dev,exec 0 0
192.168.1.234:/volume1/Documents/media /opt/paperless/media nfs vers=3,nouser,atime,auto,retrans=2,rw,dev,exec 0 0
```

And syno side 👍 

![image](https://github.com/tteck/Proxmox/assets/15791218/48e3d525-0a4b-4e48-9a76-2af8b76a1041)

It work for me , hope it help

**Comment by vsa-d** [comment]
Hello, @Arkheon and all,

I am trying same thing..... I would like to have access from my Synology nfs-Share to the Paperless archive.

- With help of the Proxmox-helper-scrips I installed paperless-ngx (no changes, standard, unprivileged lxc).  Runs perfectly.
- I did in  Paperless lxc: 
   apt update && apt install nfs-common -y
- On Synology I gave the common shares the above permission to /volume1/Proxmox/    which contains to directories (consume and media)
- in fstab I entered:
192.168.178.75:/volume1/Proxmox/consume /opt/paperless/consume nfs vers=3,nouser,atime,auto,retrans=2,rw,dev,exec 0 0
192.168.178.75:/volume1/Proxmox/media /opt/paperless/media nfs vers=3,nouser,atime,auto,retrans=2,rw,dev,exec 0 0

mount -a says "no permission"
What am I doing wrong)?
Thanks.....

**Reply by AlexGeek77** [reply]
Unprivledged containers do not have access to NFS or other remote file sharing services. You'll need to mount those drives on the Proxmox host and then pass them through to the container while also doing some user binding to enable write access.

**Reply by vsa-d** [reply]
Thanks! Is it possible to covet that lxc to privileged easily? (Proxmox with Paperless lxc is only locally in home-network, not reachable from outside.)

**Reply by AlexGeek77** [reply]
Sorry, I don't believe so.

**Reply by alexanderjerome** [reply]
yep, just edit the conf file and set unprivilidged to 0.
on the host machine, nano /etc/pve/lxc/vmid.conf I believe
